libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Conway/SimplePParams.hs

```diff
@@ -0,0 +1,548 @@
+uDropConway pp psub =
+  psub
+    { upoolVotingThresholds = pp ^. ppuPoolVotingThresholdsL
+    , udrepVotingThresholds = pp ^. ppuDRepVotingThresholdsL
+    , ucommitteeMinSize = pp ^. ppuCommitteeMinSizeL
+    , ucommitteeMaxTermLength = pp ^. ppuCommitteeMaxTermLengthL
+    , ugovActionLifetime = pp ^. ppuGovActionLifetimeL
+    , ugovActionDeposit = pp ^. ppuGovActionDepositL
+    , udRepDeposit = pp ^. ppuDRepDepositL
+    , udRepActivity = pp ^. ppuDRepActivityL
+    , uminFeeRefScriptCostPerByte = pp ^. ppuMinFeeRefScriptCostPerByteL
+    }
+
+uLiftShelley :: EraPParams era => SimplePPUpdate -> PParamsUpdate era
+uLiftShelley pps =
+  emptyPParamsUpdate
+    & ppuMinFeeAL .~ (uminFeeA pps)
+    & ppuMinFeeBL .~ (uminFeeB pps)
+    & ppuMaxBBSizeL .~ (umaxBBSize pps)
+    & ppuMaxTxSizeL .~ (umaxTxSize pps)
+    & ppuMaxBHSizeL .~ (fromIntegral <$> (umaxBHSize pps))
+    & ppuKeyDepositL .~ (ukeyDeposit pps)
+    & ppuPoolDepositL .~ (upoolDeposit pps)
+    & ppuEMaxL .~ (ueMax pps)
+    & ppuNOptL .~ (unOpt pps)
+    & ppuA0L .~ (ua0 pps)
+    & ppuRhoL .~ (urho pps)
+    & ppuTauL .~ (utau pps)
+    & ppuMinPoolCostL .~ (uminPoolCost pps)
+    & ppuMinPoolCostL .~ (uminPoolCost pps)
+
+uLiftProtVer ::
+  (EraPParams era, ProtVerAtMost era 8) => SimplePPUpdate -> PParamsUpdate era -> PParamsUpdate era
+uLiftProtVer pps pp = pp & ppuProtocolVersionL .~ (uprotocolVersion pps)
+
+uLiftAlonzo :: AlonzoEraPParams era => SimplePPUpdate -> PParamsUpdate era -> PParamsUpdate era
+uLiftAlonzo pps pp =
+  pp
+    & ppuCostModelsL .~ (ucostModels pps)
+    & ppuPricesL .~ (uprices pps)
+    & ppuMaxTxExUnitsL .~ (umaxTxExUnits pps)
+    & ppuMaxBlockExUnitsL .~ (umaxBlockExUnits pps)
+    & ppuMaxValSizeL .~ (umaxValSize pps)
+    & ppuCollateralPercentageL .~ (ucollateralPercentage pps)
+    & ppuMaxCollateralInputsL .~ (umaxCollateralInputs pps)
+
+uLiftBabbage :: BabbageEraPParams era => SimplePPUpdate -> PParamsUpdate era -> PParamsUpdate era
+uLiftBabbage pps pp = pp & ppuCoinsPerUTxOByteL .~ (CoinPerByte <$> (ucoinsPerUTxOByte pps))
+
+uLiftConway :: ConwayEraPParams era => SimplePPUpdate -> PParamsUpdate era -> PParamsUpdate era
+uLiftConway pps pp =
+  pp
+    & ppuPoolVotingThresholdsL .~ (upoolVotingThresholds pps)
+    & ppuDRepVotingThresholdsL .~ (udrepVotingThresholds pps)
+    & ppuCommitteeMinSizeL .~ (ucommitteeMinSize pps)
+    & ppuCommitteeMaxTermLengthL .~ (ucommitteeMaxTermLength pps)
+    & ppuGovActionLifetimeL .~ (ugovActionLifetime pps)
+    & ppuGovActionDepositL .~ (ugovActionDeposit pps)
+    & ppuDRepDepositL .~ (udRepDeposit pps)
+    & ppuDRepActivityL .~ (udRepActivity pps)
+    & ppuMinFeeRefScriptCostPerByteL .~ (uminFeeRefScriptCostPerByte pps)
+
+-- ============================================================================
+-- Term Selectors for SimplePParams
+
+minFeeA_ :: (EraPP era, BaseUniverse fn) => Term fn (SimplePParams era) -> Term fn Coin
+minFeeA_ simplepp = sel @0 simplepp
```

Comment:
I hope we can one day refer to the fields by their name instead of using an index. Something like `field @"minFeeA" simplepp`. That way we can be sure that we don't break anything when someone reorders some fields in the PParams.

Reply:
Yes, but I don't know how to do that. Maybe make a type family from Symbol to the type of some Lens', and then use the type family.  I will give it a try.